src/sentry/rules/processing/delayed_processing.py

```diff
@@ -432,7 +432,16 @@ def apply_delayed(project_id: int, *args: Any, **kwargs: Any) -> None:
     Grab rules, groups, and events from the Redis buffer, evaluate the "slow" conditions in a bulk snuba query, and fire them if they pass
     """
     # STEP 1: Fetch the rulegroup_to_event_data mapping for the project from redis
-    project = Project.objects.get_from_cache(id=project_id)
+    try:
+        project = Project.objects.get_from_cache(id=project_id)
+    except Project.DoesNotExist:
```

Comment:
does `get_from_cache` update the cache as well? (looking at the code, it looks like it's only updating an in memory `local_cache` and a django cache `cache`).

should we manually eject the project id from the buffer instead of relying on the TTL In redis? (my fear is that we're going to be creating 60 * projects logs per hour, until these are cleaned up in the TTL)

Reply:
I didn't realize this but we do actually delete the `project_id` after the task for that project is fired off
https://github.com/getsentry/sentry/blob/af5beb5dfc533a91485311721a7a2b5f0fbf4b3e/src/sentry/rules/processing/delayed_processing.py#L417

As for caching, if the project isn't found it doesn't update the cache. Is that what you mean by update the cache?